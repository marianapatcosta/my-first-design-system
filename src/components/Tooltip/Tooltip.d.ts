import React, { CSSProperties } from 'react';
export interface TooltipProps {
    /**
     * element id
     */
    id?: string;
    /**
     * additional css class
     */
    className?: string;
    /**
     * additional style
     */
    style?: CSSProperties;
    /**
     * Tooltip's text
     */
    text: string;
}
export declare const Tooltip: React.FC<TooltipProps>;
