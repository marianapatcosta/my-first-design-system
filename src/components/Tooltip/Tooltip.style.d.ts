export declare const StyledTooltip: import("styled-components").StyledComponent<"div", any, {}, never>;
