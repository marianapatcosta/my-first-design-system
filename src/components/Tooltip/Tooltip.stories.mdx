import { Meta } from "@storybook/addon-docs/blocks";
import { Tooltip } from "./Tooltip";

<Meta title="Design System|Tooltip" component={Tooltip} />

# Tooltip
Simple Tooltip component

## Usage
Use Tooltip when some helper text is needed; With extra styling, tooltip can be render on 
mouseover or on click of another html element

- Default Tooltip:
  <Story name='Default'>
    <Tooltip text='Tooltip'/>
  </Story>


## Other important notes
not applicable.