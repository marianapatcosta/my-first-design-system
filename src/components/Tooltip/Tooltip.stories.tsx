import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Tooltip, TooltipProps } from './Tooltip';

export default {
  title: 'Tooltip',
  component: Tooltip,
  args: {},
} as Meta;

const Template: Story<TooltipProps> = (args) => <Tooltip {...args} />;

export const Default = Template.bind({});
Default.args = { text: 'Tooltip', style: { marginLeft: '100px' } };
