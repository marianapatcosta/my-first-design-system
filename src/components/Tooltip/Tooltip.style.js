var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
import { baseStyles, colors } from '../../styles';
export var StyledTooltip = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  ", ";\n  background-color: ", ";\n  opacity: 0.8;\n  font-size: 85%;\n  padding: 0.3rem;\n  box-shadow: 0 0.125rem 0.125rem ", ";\n  bottom: -1.2rem;\n  left: 50%;\n  transform: translate(-50%, 0);\n  z-index: 2;\n  border-radius: 0.3rem;\n  -webkit-border-radius: 0.3rem;\n  -moz-border-radius: 0.3rem;\n  -ms-border-radius: 0.3rem;\n  -o-border-radius: 0.3rem;\n  width: fit-content;\n  height: fit-content;\n  white-space: nowrap;\n\n  &::after {\n    content: ' ';\n    position: absolute;\n    top: -0.4rem;\n    left: 0;\n    right: 0;\n    width: 0;\n    margin: 0 auto;\n    border: 0.4rem solid ", ";\n    border-top-color: ", ";\n    color: ", ";\n    border-bottom: 0;\n    transform: rotate(180deg) scaleX(-1);\n    -webkit-transform: rotate(180deg) scaleX(-1);\n    -moz-transform: rotate(180deg) scaleX(-1);\n    -ms-transform: rotate(180deg) scaleX(-1);\n    -o-transform: rotate(180deg) scaleX(-1);\n  }\n"], ["\n  ", ";\n  background-color: ", ";\n  opacity: 0.8;\n  font-size: 85%;\n  padding: 0.3rem;\n  box-shadow: 0 0.125rem 0.125rem ", ";\n  bottom: -1.2rem;\n  left: 50%;\n  transform: translate(-50%, 0);\n  z-index: 2;\n  border-radius: 0.3rem;\n  -webkit-border-radius: 0.3rem;\n  -moz-border-radius: 0.3rem;\n  -ms-border-radius: 0.3rem;\n  -o-border-radius: 0.3rem;\n  width: fit-content;\n  height: fit-content;\n  white-space: nowrap;\n\n  &::after {\n    content: ' ';\n    position: absolute;\n    top: -0.4rem;\n    left: 0;\n    right: 0;\n    width: 0;\n    margin: 0 auto;\n    border: 0.4rem solid ", ";\n    border-top-color: ", ";\n    color: ", ";\n    border-bottom: 0;\n    transform: rotate(180deg) scaleX(-1);\n    -webkit-transform: rotate(180deg) scaleX(-1);\n    -moz-transform: rotate(180deg) scaleX(-1);\n    -ms-transform: rotate(180deg) scaleX(-1);\n    -o-transform: rotate(180deg) scaleX(-1);\n  }\n"])), baseStyles, colors.secondary, colors.shadow, colors.transparent, colors.secondary, colors.primary);
var templateObject_1;
