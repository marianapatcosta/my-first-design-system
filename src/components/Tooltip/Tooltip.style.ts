import styled from "styled-components";
import { baseStyles, colors } from '../../styles';

export const StyledTooltip = styled.div`
  ${baseStyles};
  position: absolute;
  background-color: ${colors.highlight};
  opacity: 0.8;
  font-size: 85%;
  padding: 0.3rem;
  box-shadow: 0 0.125rem 0.125rem ${colors.shadow};
  bottom: -1.2rem;
  left: 50%;
  transform: translate(-50%, 0);
  z-index: 2;
  border-radius: 0.3rem;
  -webkit-border-radius: 0.3rem;
  -moz-border-radius: 0.3rem;
  -ms-border-radius: 0.3rem;
  -o-border-radius: 0.3rem;
  width: fit-content;
  height: fit-content;
  white-space: nowrap;

  &::after {
    content: ' ';
    position: absolute;
    top: -0.4rem;
    left: 0;
    right: 0;
    width: 0;
    margin: 0 auto;
    border: 0.4rem solid ${colors.transparent};
    border-top-color: ${colors.highlight};
    color: ${colors.primary};
    border-bottom: 0;
    transform: rotate(180deg) scaleX(-1);
    -webkit-transform: rotate(180deg) scaleX(-1);
    -moz-transform: rotate(180deg) scaleX(-1);
    -ms-transform: rotate(180deg) scaleX(-1);
    -o-transform: rotate(180deg) scaleX(-1);
  }
`;