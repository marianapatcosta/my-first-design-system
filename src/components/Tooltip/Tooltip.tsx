import React, { CSSProperties } from 'react';
import { StyledTooltip } from './Tooltip.style';

export interface TooltipProps {
  /**
   * element id
   */
  id?: string;
  /**
   * additional css class
   */
  className?: string;
  /**
   * additional style
   */
  style?: CSSProperties;
  /**
   * Tooltip's text
   */
  text: string;
}

export const Tooltip: React.FC<TooltipProps> = ({ text, ...props }) => {
  return <StyledTooltip {...props}>{text}</StyledTooltip>;
};

Tooltip.defaultProps = {
  className: '',
  style: {},
  id: '',
  text: '',
};
