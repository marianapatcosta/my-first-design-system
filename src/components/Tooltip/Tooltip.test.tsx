import React from 'react';
import { render } from '@testing-library/react';
import { Tooltip } from './Tooltip';

describe('Tooltip', () => {
  const defaultProps = {text:'Tooltip'};
   test('renders without errors and matches snapshot', () => {
     const { container: { firstChild }  } = render(<Tooltip  {...defaultProps}/>); 

     expect(firstChild).toMatchSnapshot();
   });

   test('should display text', () => {
    const { getByText } = render(<Tooltip  {...defaultProps}/>); 

    getByText(/Tooltip/i);
  });
});