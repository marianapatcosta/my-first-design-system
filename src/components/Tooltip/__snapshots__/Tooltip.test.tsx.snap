// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders without errors and matches snapshot 1`] = `
<div
  class="sc-bdfBwQ llGMZx"
  id=""
>
  Tooltip
</div>
`;
